rubix-common/src/main/java/com/qubole/rubix/common/utils/ClusterUtil.java

```diff
@@ -50,4 +64,66 @@ public static String getMasterHostname(Configuration conf)
     log.debug("No hostname found in etc/*-site.xml, returning localhost");
     return "localhost";
   }
+
+  public static synchronized Configuration applyRubixSiteConfig(Configuration conf) throws Exception
+  {
+    if (rubixSiteConfiguration == null) {
+      rubixSiteConfiguration  = new HashMap<>();
+      String configFileLocation = CacheConfig.getRubixSiteLocation(conf);
+      File configFile = new File(configFileLocation);
+      if (!configFile.exists()) {
+        log.warn(String.format("Config file does not exist on location: %s", configFileLocation));
+        if (!configFileLocation.equals(DEFAULT_RUBIX_SITE_LOCATION)) {
+          throw new IOException("Rubix config file not found on path: " + configFileLocation);
+        }
+      } else {
+        rubixSiteConfiguration = readXML(configFile);
+      }
+    }
+
+    if (!CacheConfig.isRubixSiteConfigApplied(conf)) {
+      for (Map.Entry<String, String> rubixConfig : rubixSiteConfiguration.entrySet()) {
+        conf.set(rubixConfig.getKey(), rubixConfig.getValue());
+      }
+      CacheConfig.setRubixConfigApplied(conf, true);
+    }
+    return conf;
+  }
+
+  private static Map<String, String> readXML(File file) throws Exception
+  {
+    DocumentBuilder documentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+    Document doc = documentBuilder.parse(file);
+    doc.getDocumentElement().normalize();
+    NodeList configurationNodeList = doc.getElementsByTagName("configuration");
+
+    if (configurationNodeList.getLength() == 0 || configurationNodeList.getLength() > 1) {
+      log.error(String.format("Error in Rubix site config: %s", file.getName()));
```

Comment:
Dont log in helper methods, just throw exception. Callers will log the exception